src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/SaveTempDataPropertyFilter.cs

```diff
@@ -1,14 +1,26 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Generic;
 using System.Reflection;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.Internal;
 
 namespace Microsoft.AspNetCore.Mvc.ViewFeatures.Internal
 {
-    public class SaveTempDataPropertyFilter : ISaveTempDataCallback
+    public class SaveTempDataPropertyFilter : ISaveTempDataCallback, IActionFilter
     {
-        public string Prefix { get; set; }
+        private readonly ITempDataDictionaryFactory _factory;
+
+        internal IList<PropertyHelper> PropertyHelpers { get; set; }
+
+        public SaveTempDataPropertyFilter(ITempDataDictionaryFactory factory)
+        {
+            _factory = factory;
+        }
+
+        public string Prefix = "TempDataProperty-";
```

Comment:
I do not want to block this PR for this, but could you log an issue for this as its related to user experience.